src/compiler/compile/nodes/shared/Expression.ts

```diff
@@ -391,23 +390,16 @@ export default class Expression {
 
 		if (declarations.length > 0) {
 			block.maintain_context = true;
-			declarations.forEach(declaration => {
-				block.chunks.init.push(declaration);
-			});
+			block.chunks.init.push(...declarations);
 		}
 
 		return (this.manipulated = node as Node);
 	}
 }
 
 function get_function_name(_node, parent) {
-	if (parent.type === 'EventHandler') {
-		return `${parent.name}_handler`;
-	}
-
-	if (parent.type === 'Action') {
-		return `${parent.name}_function`;
-	}
-
-	return 'func';
+	return {
+		EventHandler: `${parent.name}_handler`,
+		Action: `${parent.name}_function`
+	}[parent.type] || 'func';
```

Comment:
I don't see why this is better than the original code, it may seemed "cleaner", but you end up creating an object unnecessarily, concatenating strings for both `EventHandler` and `Action` and only using one.

the original code is simpler, cleaner, and more performant.